### ISSUE TITLE
[QUESTION] Finetuning on different model

### ISSUE THREAD

**Issue by banditelol** [koaning/icepickle#4]
Hello, I'm confused about this part of the code explaining fine-tuning on new model.
https://github.com/koaning/icepickle/blob/3472aa4bf8071b62755737d3f446d1315d459474/README.md?plain=1#L102-L112
Looking at this code example, you're doing pretraining on `LogisticRegression` model and then load the coefficient on `SGDClassifier`, I don't understand the meaning of this process. 
Do you somehow adapt the coefficient to `SGDClassifier`, or do you create a `LogisticRegression` behind the `SGDClassifier` and just train the `SGDClassifier` with the new dataset using `partial_fit`?

**Comment by koaning** [comment]
Both `SGDClassifier` and `LogisticRegression` are linear models. They train a coefficient for each input column in `X` per label, as well as an intercept per label. They really are equivalent in many ways, with the main exception that `LogisticRegression` does not implement `partial_fit`, only `SGDClassifier` does.

The "trick" that I pull off here is that I use `LogisticRegression` to pre-train on a large dataset and that I'm able to save these coefficients. These could later be loaded by another `LogisticRegression` but I can also use `SGDClassifier` because both models work in the same way. The main difference is that `SGDClassifier` can learn on micro-batches and they allow me to fine-tune on a large dataset that does not fit in memory via `partial_fit`. 

If you're unfamiliar with `partial_fit` then you may appreciate [this course](https://calmcode.io/partial_fit/introduction.html).

**Comment by banditelol** [comment]
I got the part where `SGDClassifier` implement `partial_fit`, thus I could fine-tune my logistic regression model using it. What boggle my mind is that I thought `SGDClassifier` by default implement linear `SVC` [unless I set the `loss='log'`](https://scikit-learn.org/stable/modules/sgd.html#stochastic-gradient-descent). And I haven't gotten as far as checking if the math checks out if I load `LogisticRegression` `_coef` and `_intercept` on linear `SVC` model, but for now I'm still skeptical that it's compatible.

As for the "trick" part I still don't understand why don't I just use `SGDClassifier` in the first place?

And thanks for the recommendation, I'll definitely check the course. :)

**Comment by banditelol** [comment]
On similar note, shouldn't the package also save the model's exposed parameters to make sure the loaded model still work the same way? or is it left for the user to make sure they instantiate the same model?

For example if I save `SGDClassifier` without also saving its `loss`, its `predict_proba(X)` method probably won't work as expected (I haven't tried it out myself, but looking at [the source](https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/linear_model/_stochastic_gradient.py#L1243-L1284), I assume so)

**Comment by koaning** [comment]
> As for the "trick" part I still don't understand why don't I just use SGDClassifier in the first place?

That's indeed a much better idea! Will make the edit. You're also right about `loss="log"` will add that right away!

**Comment by koaning** [comment]
Thanks for your comment! I think the example is much better now. Feel free to glance again.

**Comment by banditelol** [comment]
Great! I'll close this issue then, I'll let you know if I have another questions
Thanks for the Answer :)
